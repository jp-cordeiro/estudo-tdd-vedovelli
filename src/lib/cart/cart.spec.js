import Cart from './cart';

describe('Cart', () => {
  let cart;

  beforeEach(() => {
    cart = new Cart();
  });

  it('return 0 when getTotal() is executed in a newly created instance', () => {
    expect(cart.getTotal()).toEqual(0);
  });

  it('multiply quantity and price and receive the total amount ', () => {
    const item = {
      product: {
        title: 'Action Figure Luffy',
        price: 35388,
      },
      quantity: 2,
    };

    const totalExpected = 2 * 35388;

    cart.addItem(item);
    expect(cart.getTotal()).toEqual(totalExpected);
  });
});
