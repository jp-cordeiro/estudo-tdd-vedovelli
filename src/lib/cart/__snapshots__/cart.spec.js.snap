// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart checkout() return an object with the total and the list items 1`] = `
Object {
  "items": Array [
    Object {
      "product": Object {
        "price": 35388,
        "title": "Action Figure Luffy",
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "price": 12345,
        "title": "Action Figure Zoro",
      },
      "quantity": 3,
    },
  ],
  "total": 107811,
}
`;

exports[`Cart checkout() return an object with the total and the list items when sumary is called 1`] = `
Object {
  "items": Array [
    Object {
      "product": Object {
        "price": 35388,
        "title": "Action Figure Luffy",
      },
      "quantity": 2,
    },
    Object {
      "product": Object {
        "price": 12345,
        "title": "Action Figure Zoro",
      },
      "quantity": 3,
    },
  ],
  "total": 107811,
}
`;
